app/components/RouteModal/RouteModal.tsx

```diff
@@ -0,0 +1,58 @@
+import {
+  Box,
+  CloseButton,
+  Container,
+  Flex,
+  LoadingOverlay,
+  Text,
+  Tooltip,
+} from "@mantine/core"
+import { FetcherWithComponents, NavLink } from "@remix-run/react"
+import React, { ReactNode } from "react"
+import PortalLoader from "~/components/PortalLoader"
+
+type RouteModalProps = {
+  state: FetcherWithComponents<any>["state"]
+  children: ReactNode
+  loaderMessage: string
+}
+
+const RouteModal = ({ state, children, loaderMessage }: RouteModalProps) => {
+  return state === "idle" ? (
+    <Container p="xl" size="md">
+      <Box mt={{ base: 0, sm: 90 }}>{children}</Box>
+    </Container>
+  ) : (
+    <LoadingOverlay
+      visible
+      loaderProps={{
+        children: <PortalLoader message={loaderMessage} />,
+      }}
+    />
+  )
+}
+
+type RouteModalHeaderProps = {
+  title: string
+  description?: string
+  closeButtonLink: string
+}
+const Header = ({ title, description, closeButtonLink }: RouteModalHeaderProps) => (
```

Comment:
For being components in the same page, these dont ever seem to be used at the same time. Like Header is never used inside of RouteModal. So then, I'm not really sure why they in the same file.

Reply:
It's defined here to be used as a compound component that comes from the `RouteModal`. 

See this line 
https://github.com/pokt-foundation/portal-platform/blob/4aa69ee25fe6321cb441a579d0155ac3580bb706/app/components/RouteModal/RouteModal.tsx#L56

Reply:
Yeah I followed that. But typically compound components are used together. Like Tab and Tab.Panel. To use Tab.Panel you almost always have to use Tab. But you are often using RouteModal.Header without RouteModal.

Reply:
This is more of a clarification on when this should be used, you can use it anywhere you want (but you can't expect it to look/behave as it should), that applies to almost any compound component, and in our case it has to be used inside of the `RouteModal`. 
In all of our cases, it's used within a component that's a direct child of the `RouteModal`. 

Same thing can be found in Mantine's AppShell for example.. this is how we're using the `AppShell.Navbar` component inside of a direct child `AppShell`. 

https://github.com/pokt-foundation/portal-platform/blob/4aa69ee25fe6321cb441a579d0155ac3580bb706/app/components/Sidebar/Sidebar.tsx#L76